src/libsystemd/sd-bus/bus-introspect.c

```diff
@@ -86,7 +86,7 @@ static void introspect_write_flags(struct introspect *i, int type, uint64_t flag
                 fputs("   <annotation name=\"org.freedesktop.systemd1.Privileged\" value=\"true\"/>\n", i->f);
 }
 
-static int introspect_write_arguments(struct introspect *i, const char *signature, const char *direction) {
+static int introspect_write_arguments(struct introspect *i, const char *signature, const char **names, const char *direction) {
```

Comment:
so usually we have input params or output params. But *names in this case is both. Since that is untypical and thus unexpected this really should have a comment here somewhere. just a brief line like this suffices:

```c
/* Note that "names" is both an input and an output parameter. It initially points to the first argument name in a NUL seperated list of strings, and is then advanced with each argument, and the resulting pointer returned. */
```

Or something along those lines